pkg/server/handler/v1alpha1/template.go

```diff
@@ -45,7 +54,59 @@ func ListTemplates(ctx context.Context, tenant string, includePublic bool, query
 		items = append(items, publicTemplates.Items...)
 	}
 
-	size := int64(len(items))
+	results := []v1alpha1.Stage{}
+	if query.Filter == "" {
+		results = items
+	} else {
+		// Support multiple filters rules: name or alias, and type, separated with comma.
+		filterParts := strings.Split(query.Filter, ",")
+		filters := make(map[string]string)
+		for _, part := range filterParts {
+			kv := strings.Split(part, "=")
+			if len(kv) != 2 {
+				return nil, cerr.ErrorQueryParamNotCorrect.Error(query.Filter)
+			}
+
+			filters[kv[0]] = strings.ToLower(kv[1])
+		}
+
+		var selected bool
+		for _, item := range items {
+			selected = true
+			for key, value := range filters {
+				switch key {
+				case "name":
+					if !strings.Contains(item.Name, value) {
+						selected = false
+					}
+				case "alias":
+					if item.Annotations != nil {
+						if alias, ok := item.Annotations[meta.AnnotationAlias]; ok {
```

Comment:
If user specified `alias` filter, but a item doesn't have annotation alias, will it be accepted ? I think here if `item.Annotations` is nil or `ok` is false, also need set `selected` to false.